### PR TITLE
32 add dockerfiles

### PR DESCRIPTION
**Pull request by TwelveNights**
Had to re-create #37 due to the branch name violating git prehooks. The PR resolves some of the review comments.

### CODE REVIEW

**Comment by yann510** [comment]
> Had to re-create #37 due to the branch name violating git prehooks. The PR resolves some of the review comments.

Hmmm yeah, that's a problem. We should validate the branch name directly upon its creation.

File: .github/workflows/build-docker-webapp.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        env:
+          nx_cloud_token: ${{ secrets.NX_CLOUD_TOKEN }}
+        run:
+          sed -i "s/nx_cloud_token/$nx_cloud_token/" $GITHUB_WORKSPACE/nx.json
+
+      - name: Install npm packages
+        run: npm ci
+
+      - name: Build webapp
+        run: nx build api
```
Comment (GabrielBottari) [inline .github/workflows/build-docker-webapp.yml:40]: typo, you're building the api twice

**Review by GabrielBottari — COMMENTED** [review]
Aren't you also supposed to add a dispatch call somewhere? I see two cases:
1- We want to deploy on merge with master, so dispatch should be called after the main CI succeeds
2- We want to produce preview apps, so they should probably be run on PR instead, not sure if there will be differences

File: .github/workflows/build-docker-webapp.yml (hunk 1 above)
Comment (TwelveNights) [reply .github/workflows/build-docker-webapp.yml:40]: argh good catch

**Comment by TwelveNights** [comment]
> Aren't you also supposed to add a dispatch call somewhere? I see two cases:
> 1- We want to deploy on merge with master, so dispatch should be called after the main CI succeeds
> 2- We want to produce preview apps, so they should probably be run on PR instead, not sure if there will be differences

For now I wanted to set it to manual so that we don't needlessly build images. Or if we do, we should use something other than GitHub packages.

**Review by GabrielBottari — APPROVED** [review]

**Pull request merged**